src/WidgetBase.ts

```diff
@@ -409,11 +437,22 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 			const render = beforeRenders.reduce((render, beforeRenderFunction) => {
 				return beforeRenderFunction.call(this, render, this._properties, this._children);
 			}, this.render.bind(this));
+
+			this._requiredNodes.forEach((element, key) => {
```

Comment:
I think it's not a private function because it's so similar to the code structure of the surrounding beforeRender/afterRenders calls right before and after it. It could certainly be a private function and, if so, should I do the same thing with the code around it?

I moved it before beforeRender and it works fine.